### ISSUE TITLE
[Bug]: Poisonous Mushrooms causing entity lag due to potion AoE effects

### ISSUE THREAD

**Issue by katubug** [cech12/ExtendedMushrooms#86]
### Minecraft Version

1.19.2

### Forge Version

43.3.5

### Mod Version

3.0.16

### Modpack or other Mod

It's in the Cottage Witch modpack: https://www.curseforge.com/minecraft/modpacks/cottage-witch

### Describe the Issue

We're troubleshooting some lag issues, and found that when Poisonous Mushrooms are in loaded chunks, they constantly spawn potion effect entities: https://i.imgur.com/rhNZao0.png

This is not particularly performant, especially on large servers.

### Logs

I don't have access to the logs at this time.

### Do you use any performance-enhancing mods (e.g. OptiFine)?

Yes, but not optifine.

**Comment by cech12** [comment]
Hey thanks for your report. But this is a feature of this mod. ^^ What is the expected behavior? Could you add the performance enhancing mods?

**Comment by alltheArdy** [comment]
(I am an admin on the mentioned server)

So the big issue is that this occurred while there was no player right next to it. (they were a decent bit away)

 It also is extremely excessive and there is no configuration for it in the files.

![image](https://github.com/cech12/ExtendedMushrooms/assets/138737686/24cfe9f8-abe3-429a-8849-2f819184e611)

regarding performance mods, we have servercore, apt, canary as the ones off the top of my head.

Basically the effect generates far too much, and takes a long time to fade away as well.

**Comment by cech12** [comment]
Wow, that is too much! The normal/expected amount should be 1-3 clouds around a mushroom... Maybe one of these performance mods are causing an incompatibility and let the mushrooms tick too much. I will look into that, when I have time :) Thanks for your detailed report!

**Comment by cech12** [comment]
@alltheArdy @katubug 
I have tested it and I cannot find an issue. If only Extended Mushrooms is installed, it looks like that:

![Image](https://github.com/cech12/ExtendedMushrooms/assets/16005054/b4b71e8a-2587-4cda-a6dd-8a6e7488120f)

And when I test it in the mod pack Cottage Witch it looks like that:

![Image](https://github.com/cech12/ExtendedMushrooms/assets/16005054/c427994f-8d51-4998-ab97-e573d9757014)

In both tests I placed 4 big poisonous mushrooms and let it run for a while. There was no moment where more than 5 effect entities were there.

Maybe you have a higher randomTickSpeed? (the default value is 3) Is it higher, more effect entities are added over time.

**Comment by cech12** [comment]
I have added a config option for balancing the effect clouds. 
It is available in the upcoming versions 1.19.2-3.0.2.0, 1.19.4-3.2.1.0 and 1.20.1-4.1.2.0.
I hope, this helps you in your environment. Please let me know, if this issue is fixed with that config option and when you have other issues. :)

Thanks again for your report. :)

**Comment by cech12** [comment]
There was no response, so I assume that the change had the desired effect. Therefore, the issue is closed.
If there are still problems, please open the issue with new information.
Thanks again for the report!

**Comment by katubug** [comment]
Ah yes, sorry about that! I was waiting to hear back from my team and then
totally forgot to respond. The config is good!

**Comment by cech12** [comment]
Strangely, it looks to me as if you have configured a higher tick rate. This could also cause problems in other contexts. But it's nice to hear that the config option here solved the problem for you. :)

**Comment by katubug** [comment]
I'll be honest, I haven't personally been able to test it yet. The server which was having issues hasn't yet updated to the latest modpack version. It's very possible that we misunderstood the config and thought we were increasing time between ticks? I'll check into it, thank you for letting me know!

**Comment by cech12** [comment]
The new config option does not increase the tick rate. You can tweak the chance of generating effect clouds at a random tick of a mushroom cap block. (default is 5%). It has a good behaviour (see my screenshots above) for the default random tick rate of 3 (vanilla setting). If you double to random tick rate to a value of 6 you should half the new config option to a value of 2.5 % to have the same behaviour.

But I would expect that you increased the random tick rate to find possible lag sources, which causes the high amount of effect clouds. If you decrease the random tick rate to the default of 3 you don't have to tweak the new option :)

**Comment by katubug** [comment]
Ah, now I understand! I am not in charge of the server we were reporting about, that's why I was confused. @alltheArdy Can you confirm you had turned the tick rate up on the server?

That makes much more sense, now. We will adjust the config accordingly to the server I DO run, then. Thank you! :)
